xbmc.py

```diff
@@ -250,24 +250,230 @@ def __getitem__(self, i):
 
 # noinspection PyPep8Naming,PyArgumentList
 class Player(KodiStub):
-    playing_file = None
+    _STATUS_INIT = 'init'
+    _STATUS_STOPPED = 'stopped'
+    _STATUS_PLAYING = 'playing'
+    _STATUS_PAUSED = 'paused'
+
+    _stop_event = threading.Event()
+
+    __instance = None
+    __status = _STATUS_STOPPED
+    __file = None
+    __current_time = 0
+    __total_time = 0
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self):
         super(Player, self).__init__()
 
-        self.Init(*args, **kwargs)
-        self.play_started = False
+        # Keep track of the last so we can notify it when something happens.
+        Player.__instance = self
+
+    @classmethod
```

Comment:
Why all the `@classmethod` methods? The result is a class that mixes class methods and normal instance methods. The Player is an actual object in Kodi, so I don't really prefer the class methods here. Could you change all of these to 'normal' instance methods?

Reply:
It's my attempt to create a singleton Player. You can start playback trough setResolvedStream, and then instanciate a Player object, and that should be playing.